scripts/lib/CIME/XML/generic_xml.py

```diff
@@ -310,6 +329,9 @@ def write(self, outfile=None, force_write=False):
         else:
             with open(outfile,'w') as xmlout:
                 xmlout.write(xmlstr)
+
+        self._FILEMAP[self.filename] = (self.tree, self.root, os.path.getmtime(self.filename))
```

Comment:
In theory, it seems like this could be vulnerable to the same race condition described above, but I don't know how to solve this one... and hopefully the system would prevent two processes from trying to write to the same file at the same time anyway, so the risk here seems small enough.